integration_tests/tests/sns-lambda-tests.ts

```diff
@@ -28,30 +28,32 @@ async function waitForSubstringInLogs(
 
 describe(
   "\n Happy path tests \n" +
-    "\n publish valid sns message and check cloud watch logs lambda functions Filter,Clean, Store Transactions contains eventId\n",
+    "\n Generate valid event and check cloud watch logs lambda functions Filter,Clean, Store Transactions contains eventId\n",
   () => {
     beforeAll(async () => {
-      await publishToTestTopic(snsValidEventPayload);
+      await generateEventViaFilterLambdaAndCheckEventInS3Bucket(
```

Comment:
Similar to my previous suggestions, you could use the eventId that is being returned by the `generateEventViaFilterLambdaAndCheckEventInS3Bucket` function. 
```suggestion
    let eventId: string;

    beforeAll(async () => {
      const result = await generateEventViaFilterLambdaAndCheckEventInS3Bucket(
        validEventPayload
      );
      if (result.eventId) {
        eventId = result.eventId;
      }
    });

    test("Filter function cloud watch logs should contain eventid", async () => {
      await waitForSubstringInLogs("-filter-function", eventId);
    });

    test("Clean function cloud watch logs should contain eventid", async () => {
      await waitForSubstringInLogs("-clean-function", eventId);
    });

    test("Store Transactions function cloud watch logs should contain eventid", async () => {
      await waitForSubstringInLogs("-storage-function", eventId);
    });
```

Reply:
updated as suggested